### DISCUSSION TITLE
[Bug]: Different values between FusionSolar App and this Integration

### DISCUSSION THREAD

**Discussion by dnnrwttr** [wlcrs/huawei_solar#788] [General]
### Describe the issue

Hi ive got different values in FusionSolar App for example 40kwh daily yield and in that Integration just 34.

An reason for that?!

And would it be possible to get the value of "used from grid" from the smart Meter?!

### Describe your Huawei Solar Setup

Inverter Type: 
Inverter Firmware version: 
sDongle present: Yes / No
sDongle Type: sDongleA-05 (WiFi / Ethernet) / SDongleA-03 (4G)
sDongle Connectivitiy: WiFi / Ethernet / 4G
sDongle Firmware: 
Power meter present: three phase / single phase / no
Optimizers Present: Yes / No
Battery: LUNA2000-SO xxkWh / LUNA2000-S1 xxkWh (Released 2024) / LG RESU xxkWh / None
Battery Firmware version: 
Huawei Solar integration version: 

### How do you connect to the inverter?

Please select your connection method

### Upload your Diagnostics File

Drag & Drop your Diagnostics File here.

### Upload your relevant debug logs

```text
.
```

### Please confirm the following:

- [ ] I'm running the latest release of Home Assistant.
- [ ] I'm running the latest release of this integration.
- [ ] I did not find an existing issue describing this problem.
- [ ] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [ ] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by Roving-Ronin** [comment]
Huawei integration reads the modbus registers on the Inverters/Meters/Battery etc... Running 3 different power monitoring systems (IoTaWatt , SolarAnalytics and this integration) I would accept the values this integration gives, before the values in FusionSolar (that only updates every 5 or was it 15 minutes and has bugs here and there). Also SF is not one 'global' installation its a bunch of servers dotted all over the world to which users based on location etc are sent to different physical servers. Software isn't Huawei's either it is developed by a company in Singapore and Huawei license its usage.

"used from grid"... that would already be there, IF you have a consumption meter installed (which we can't tell as you didn't bother to define anything in the "Describe your Huawei Solar Setup" section). Assuming you have a meter lookup the sensor.power_meter_consumption for the total (lifetime) value of imports from the grid.

If you want daily import values, then use the Home Assistant 'utility_meter' component and define it to report on daily frequency (and add tariff periods also etc) and use the above sensor as its data source.
